tony-history-server/app/controllers/JobsMetadataPageController.java

```diff
@@ -49,43 +49,33 @@ public JobsMetadataPageController() {
   }
 
   private void moveIntermToFinished(FileSystem fs, HdfsConfiguration conf, Map<String, Date> jobsAccessTime,
-      Map<String, String[]> jobFiles) {
+      Map<String, Path> jobFolders) {
     jobsAccessTime.forEach((id, date) -> {
       StringBuilder path = new StringBuilder(finished.toString());
       LocalDate ldate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
       String[] directories = {Integer.toString(ldate.getYear()), Integer.toString(ldate.getMonthValue()),
-          Integer.toString(ldate.getDayOfMonth()), id};
+          Integer.toString(ldate.getDayOfMonth())};
       for (String dir : directories) {
         path.append("/").append(dir);
         Utils.createDir(fs, new Path(path.toString()), Constants.perm770);
       }
 
+      Path source = jobFolders.get(id);
       Path dest = new Path(path.toString());
-      for (String f : jobFiles.get(id)) {
-        Path jobFile = new Path(f);
-        try {
-          FileUtil.copy(fs, jobFile, fs, dest, true, conf);
-        } catch (IOException e) {
-          LOG.error("Failed to move files from intermediate to finished");
-        }
+      try {
+        FileUtil.copy(fs, source, fs, dest, true, conf);
```

Comment:
It looks like you're still doing a copy. Let's use a `FileSystem.rename()` instead.

Reply:
Yep fixed.